cypress/integration/channel/channel_settings_spec.js

```diff
@@ -0,0 +1,48 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint-disable max-nested-callbacks */
+describe('Channel Settings', () => {
+    before(() => {
+        // 1. Go to Main Channel View with "user-1"
+        cy.toMainChannelView('user-1');
+    });
+
+    it('C15052 All channel types have appropriate close button', () => {
+        cy.get('#publicChannel').parent().parent().children('li:not(:first-child)').not('#morePublicButton').each(($el) => {
+            cy.wrap($el).find('a').find('span.btn-close').should('have.length', 0);
+        });
+
+        cy.get('#privateChannel').parent().parent().children('li:not(:first-child)').each(($el) => {
+            cy.wrap($el).find('a').find('span.btn-close').should('have.length', 0);
```

Comment:
Try and check for existence/visiblilty vs length or count in this kind of scenario (like in my above example).